cmd/finch/version.go

```diff
@@ -4,27 +4,107 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
 
 	"github.com/spf13/cobra"
 
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/flog"
+	"github.com/runfinch/finch/pkg/lima"
 	"github.com/runfinch/finch/pkg/version"
 )
 
-var finchVersion = version.Version
-
-func newVersionCommand() *cobra.Command {
+func newVersionCommand(limaCmdCreator command.LimaCmdCreator, logger flog.Logger, stdOut io.Writer) *cobra.Command {
 	versionCommand := &cobra.Command{
 		Use:   "version",
 		Args:  cobra.NoArgs,
 		Short: "Shows Finch version information",
-		RunE:  versionAction,
+		RunE:  newVersionAction(limaCmdCreator, logger, stdOut).runAdapter,
 	}
 
 	return versionCommand
 }
 
-func versionAction(cmd *cobra.Command, args []string) error {
-	fmt.Println("Finch version:", finchVersion)
+type versionAction struct {
+	creator command.LimaCmdCreator
+	logger  flog.Logger
+	stdOut  io.Writer
+}
+
+// NerdctlVersionOutput captures the nerdctl version.
+type NerdctlVersionOutput struct {
+	Client NerdctlClientOuput  `json:"Client"`
+	Server NerdctlServerOutput `json:"Server"`
+}
+
+// NerdctlClientOuput captures the nerdctl Client output.
+type NerdctlClientOuput struct {
+	Version    string                    `json:"Version"`
+	GitCommit  string                    `json:"GitCommit"`
+	GoVersion  string                    `json:"GoVersion"`
+	Os         string                    `json:"Os"`
+	Arch       string                    `json:"Arch"`
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlServerOutput captures the nerdctl Server output.
+type NerdctlServerOutput struct {
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlComponentsOutput captures the nerdctl components output.
+type NerdctlComponentsOutput struct {
+	Name    string `json:"Name"`
+	Version string `json:"Version"`
+	Details struct {
+		GitCommit string `json:"GitCommit"`
+	}
+}
+
+func newVersionAction(creator command.LimaCmdCreator, logger flog.Logger, stdOut io.Writer) *versionAction {
+	return &versionAction{creator: creator, logger: logger, stdOut: stdOut}
+}
+
+func (va *versionAction) runAdapter(cmd *cobra.Command, args []string) error {
+	return va.run()
+}
+
+func (va *versionAction) run() error {
+	status, err := lima.GetVMStatus(va.creator, va.logger, limaInstanceName)
+	if (status != lima.Running) || (err != nil) {
+		fmt.Println("Finch version:", version.Version)
+		fmt.Println("Please run finch vm start to start the virtual machine")
```

Comment:
Also, we can print different things depending on the status. If it's nonexistent, then "vm init"; if it's stopped, then "vm start".

```suggestion
		fmt.Println(`Please run "finch vm start" to start the virtual machine to get the version of other components (e.g., nerdctl)`)
```

Reply:
I would rather not complicate the command and print different status here, which is already being implement in `finch vm status`. I will just remove this line